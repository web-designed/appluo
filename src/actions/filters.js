//*******************************************************
// FILTER ACTION GENERATORS
//*******************************************************

   const setTextFilter = (text) => ({
      type: 'SET_TEXT_FILTER',
      text
   })

   const sortAsc = () => ({
      type: 'SORT_ASCENDING'
   })

   const sortDesc = () => ({
      type: 'SORT_DESCENDING'
   })

   const filterByName = (name) => ({
      type: 'FILTER_BY_NAME',
      name
   })

   const filterByPlace = (place) => ({
      type: 'FILTER_BY_PLACE',
      place
   })

   const setStartDate = (date) => ({
      type: 'SET_START_DATE',
      date
   })

   const setEndDate = (date) => ({
      type: 'SET_END_DATE',
      date
   })

   export { setTextFilter, sortAsc, sortDesc, filterByName, filterByPlace, setStartDate, setEndDate }