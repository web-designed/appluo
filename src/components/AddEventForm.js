import React from 'react'
import { connect } from 'react-redux'
import uuid from 'uuid'
import moment from 'moment'
import 'react-dates/initialize'
import { SingleDatePicker } from 'react-dates'
import { Link } from 'react-router-dom'

export class AddEventForm extends React.Component {

   state = {
      date: this.props.cleanedAt ? moment(this.props.cleanedAt) : moment(),
      focused: false,
      note: this.props.note ? this.props.note : '',
      place: this.props.place ? this.props.place : this.props.settings.places[0],
      error: '',
      cleaner: this.props.cleaner ? this.props.cleaner : this.props.settings.cleaners[0],
   }

   onCleanerChange = (e) => {
      const cleaner = e.target.value
      this.setState(() => ({ cleaner }))
   }

   onPlaceChange = (e) => {
      const place = e.target.value
      this.setState(() => ({ place }))
   }

   onNoteChange = (e) => {
      const note = e.target.value
      this.setState(() => ({ note }))
   }

   onDateChange = (date) => {
      this.setState(() => ({ date }))
   }

   onFocusChange = ({focused}) => {
      this.setState(() => ({focused}))
   }

   handleSubmit = (e) => {

      e.preventDefault()
      
      const event = {
         cleanedAt: this.state.date.valueOf(),
         createdAt: moment().valueOf(),
         cleaner: this.state.cleaner,
         place: this.state.place,
         note: this.state.note,
      }

      this.props.handleSubmit(event)
   }

   render(){

      console.log(this.props)

      return(
         <div>
            <form onSubmit={this.handleSubmit}>
               <div class="input-group mb-2">
                  <div class="input-group-prepend">
                     <label class="input-group-text" for="cleaner">Name</label>
                  </div>
                  <select 
                     class="custom-select" 
                     id="cleaner"
                     name="cleaner" 
                     onChange={this.onCleanerChange}
                     value={this.state.cleaner.toLowerCase()}
                  >
                     {
                        this.props.settings.cleaners.map( cleaner => {
                           return <option value={cleaner.toLowerCase()} key={cleaner.toLowerCase()}>{cleaner}</option>
                        })
                     }
                  </select>
               </div>
               <div class="input-group mb-2">
                  <div class="input-group-prepend">
                     <label class="input-group-text" for="place">Place</label>
                  </div>
                  <select 
                     class="custom-select" 
                     id="place"
                     name="place" 
                     onChange={this.onPlaceChange}
                     value={this.state.place.toLowerCase()}
                  >
                     {
                        this.props.settings.places.map( place => {
                           return <option value={place.toLowerCase()} key={place}>{place}</option>
                        })
                     }
                  </select>
               </div>
               <div class="input-group mb-2">
                  <div class="input-group-prepend">
                     <label class="input-group-text" for="date">Date</label>
                  </div>
                  <SingleDatePicker 
                     date={this.state.date}
                     onDateChange={this.onDateChange}
                     focused={this.state.focused}
                     onFocusChange={this.onFocusChange}
                     numberOfMonths={1}
                     isOutsideRange={() => false}
                  />
               </div>
               <div class="form-group">
                  <textarea 
                     class="form-control"
                     value={this.state.note}
                     name="note" 
                     placeholder="Add your note"
                     onChange={this.onNoteChange}
                  ></textarea>
               </div>
               <div class="form-group text-right">
                  <Link class="text-danger mr-2" to={'/'}>Cancel</Link>
                  <button class="btn btn-primary">Submit</button>
               </div>
            </form>
         </div>
      )
   }
}

const mapStateToProps = ({ settings }) => ({
   settings: settings
})

export default connect(mapStateToProps)(AddEventForm)