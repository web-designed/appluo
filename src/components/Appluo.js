import React from 'react'
import EventsList from './EventsList'

class Appluo extends React.Component {
   render() {
      return (
         <div>
            <EventsList />
         </div>
      )
   }
}

export default Appluo