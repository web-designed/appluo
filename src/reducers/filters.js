//*******************************************************
// FILTERS REDUCER
//*******************************************************

   import moment from 'moment'

   const filtersReducerDefaultState = {
      text: '',
      filterByName: 'all',
      filterByPlace: 'all',
      sort: 'DESC',
      startDate: moment().startOf('month'),
      endDate: moment().endOf('month')
   }

   const filtersReducer = (state = filtersReducerDefaultState, action) => {
      switch(action.type){
         case 'SET_TEXT_FILTER':
            return {
               ...state,
               text: action.text
            }
         case 'SORT_ASCENDING':
            return {
               ...state,
               sort: 'ASC'
            }
         case 'SORT_DESCENDING':
            return {
               ...state,
               sort: 'DESC'
            }
         case 'FILTER_BY_NAME':
            return {
               ...state,
               filterByName: action.name
            }
         case 'FILTER_BY_PLACE':
            return {
               ...state,
               filterByPlace: action.place
            }
         case 'SET_START_DATE':
            return {
               ...state,
               startDate: action.date
            }
         case 'SET_END_DATE':
            return {
               ...state,
               endDate: action.date
            }
         default: 
            return state
      }
   }

   export default filtersReducer