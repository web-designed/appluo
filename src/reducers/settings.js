import moment from 'moment'

//*******************************************************
// SETTINGS REDUCER
//*******************************************************

   const settingsReducerDefaultState = {
      wgName: 'The Cleaning Team',
      slogan: 'Manage your cleaning plan',
      places: ['Kitchen', 'Bathroom', 'Saloon'],
      cleaners: ['John', 'Alex', 'Tom', 'Steve'],
      createdAt: moment('2018-12-01'),
      notifications: {
         event: {
            afterDelete: 'The cleaning was successfully removed',
            notFound: "The cleaning doesn't exist or has been removed"
         },
         comment: {
            validate: 'Please add your comment'
         }
      }
   }
   const settingsReducer = (state = settingsReducerDefaultState, action) => {
      switch(action.type){
         default:
            return state
      }
   } 

   export default settingsReducer