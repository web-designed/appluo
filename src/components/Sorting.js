import React from 'react'
import { filterByName, filterByPlace, sortAsc, sortDesc, setEndDate, setStartDate } from '../actions/filters'
import { connect } from 'react-redux'
import moment from 'moment'
import { DateRangePicker } from 'react-dates'
import uuid from 'uuid'

export class Sorting extends React.Component {

   state = {
      focusedInput: null
   }

   onDatesChange = ({ startDate, endDate }) => {
      // this line handles the case of selecting endDate that is before the startDate
      // in this case the datePicker returns null, which would crash the programm
      endDate = endDate === null ? this.props.filters.endDate : endDate
      // startDate = startDate === null ? this.props.settings.createdAt : startDate

      this.props.setEndDate(endDate)
      this.props.setStartDate(startDate)
   }

   onFocusChange = (focusedInput) => {
      this.setState(() => ({ focusedInput }))
   }

   // handleTextFilter = (e) => {
   //    this.props.dispatch(setTextFilter(e.target.value))
   // }

   handleFilterByName = (e) => {
      this.props.filterByName(e.target.value)
   }

   handleFilterByPlace = (e) => {
      this.props.filterByPlace(e.target.value)
   }

   handleSorting = (e) => {
      e.target.value === 'asc' ? this.props.sortAsc() : this.props.sortDesc()
   }

   render(){

      const hideSort = this.props.hideFilters.indexOf('sort') !== -1

      return (
         <div>
            {
            // <input 
            //    type="text" 
            //    placeholder="search" 
            //    value={this.props.filters.text} 
            //    onChange={this.handleTextFilter} 
            // />
            }
            <h6>Filter & Sort</h6>
            <div class="row">
               {
                  !hideSort && (
                     <div class="input-group col-md-3 col-lg-2 mb-3">
                        <div class="input-group-prepend">
                           <label class="input-group-text" for="sorting">Sort</label>
                        </div>
                        <select 
                           name="sorting"
                           onChange={this.handleSorting}
                           class="custom-select" 
                           id="sorting"
                           value={this.props.filters.sort.toLowerCase()} 
                        >
                           <option value="asc">ASC</option>
                           <option value="desc">DESC</option>
                        </select>
                     </div>
                  )
               }
               <div class="input-group col-lg-3 mb-3">
                  <div class="input-group-prepend">
                     <label class="input-group-text" for="place">Place</label>
                  </div>
                  <select 
                     name="place"
                     onChange={this.handleFilterByPlace}
                     class="custom-select" 
                     id="place"
                  >
                     <option value="all">All</option>
                     {this.props.settings.places.map((place, index) => {
                        return (
                           <option 
                              value={place.toLowerCase()} 
                              key="index"
                              selected = {place.toLowerCase() === this.props.filters.filterByPlace}
                           >
                              {place}
                           </option>
                        )
                     })}
                  </select>
               </div>
               
               <div class="input-group col-lg-2 mb-3">
                  <div class="input-group-prepend">
                     <label class="input-group-text" for="cleaners">Cleaners</label>
                  </div>
                  <select 
                     onChange={this.handleFilterByName}
                     name="cleaner"
                     id="cleaners"
                     class="custom-select"
                  >
                     <option value="all">All</option>
                     {this.props.settings.cleaners.map((cleaner, index) => {
                        return (
                           <option 
                              value={cleaner.toLowerCase()} 
                              key={index}
                              selected = {cleaner.toLowerCase() === this.props.filters.filterByName}
                           >
                              {cleaner}
                           </option>
                        )
                     })}
                  </select>
               </div>

               <div class="input-group col-md-9 col-lg-5 mb-3">
                  <div class="input-group-prepend">
                     <label class="input-group-text" for="cleaners">Date</label>
                  </div>
                  <DateRangePicker 
                     startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
                     startDateId={'startDateId'}
                     endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
                     endDateId={'endDateId'}
                     onDatesChange={this.onDatesChange} // PropTypes.func.isRequired,
                     focusedInput={this.state.focusedInput} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
                     onFocusChange={this.onFocusChange} // PropTypes.func.isRequired,
                     numberOfMonths = {1}
                     isOutsideRange = {() => false}
                     showClearDates = {true}
                  />
               </div>
            </div> 
         </div>
      )
   }
}

const mapDispatchToProps = (dispatch) => ({
   setEndDate: (endDate) => dispatch(setEndDate(endDate)),
   setStartDate: (startDate) => dispatch(setStartDate(startDate)),
   filterByName: (name) => dispatch(filterByName(name)),
   filterByPlace: (place) => dispatch(filterByPlace(place)),
   sortAsc: () => dispatch(sortAsc()),
   sortDesc: () => dispatch(sortDesc())
   
})

const mapStateToProps = ({ filters, settings }) => ({
   filters: filters,
   settings: settings
})

export default connect(mapStateToProps, mapDispatchToProps)(Sorting)