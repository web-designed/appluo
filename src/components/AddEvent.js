import React from 'react'
import AddEventForm from './AddEventForm'

const AddEvent = () => {
   return(
      <div>
         <h1>Add a cleaning event</h1>
         <AddEventForm />
      </div>
   )
}

export default AddEvent