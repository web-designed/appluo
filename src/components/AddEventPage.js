import React from 'react'
import AddEventForm from './AddEventForm'
import { addEvent } from '../actions/events'
import { connect } from 'react-redux'

const AddEventPage = (props) => (
   <div>
      <h1>Add a cleaning</h1>
      <AddEventForm 
         handleSubmit={ (event) => {
            props.onSubmit(event)
            props.history.push('/')
         }} 
         history={props.history} 
      />
   </div>
)

const mapDispatchToProps = (dispatch) => ({
   onSubmit: (expense) => dispatch(addEvent(expense))
})

export default connect(undefined, mapDispatchToProps)(AddEventPage)