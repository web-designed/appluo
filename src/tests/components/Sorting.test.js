import React from 'react'
import { shallow } from 'enzyme'
import { Sorting } from '../../components/Sorting'
import settings from '../fixtures/settings'
import { filters, altFilters } from '../fixtures/filters'
import moment from 'moment'

let wrapper, setEndDate, setStartDate, filterByName, filterByPlace, handleSorting, sortAsc, sortDesc

beforeEach(() => {
   setEndDate = jest.fn()
   setStartDate = jest.fn()
   filterByName = jest.fn()
   filterByPlace = jest.fn()
   handleSorting = jest.fn()
   sortAsc = jest.fn()
   sortDesc = jest.fn()
   wrapper = shallow(
      <Sorting 
         setStartDate = { setStartDate }
         setEndDate = { setEndDate } 
         filterByName = { filterByName }
         filterByPlace = { filterByPlace }
         sortAsc = { sortAsc }
         sortDesc = { sortDesc }
         settings = { settings } 
         filters = { filters } 
         hideFilters = { [] } 
      />
   )
})

test('should render Sorting correctly with default filters', () => {
   expect(wrapper).toMatchSnapshot()
})

test('should render Sorting correctly with altered filters', () => {
   wrapper.setProps({
      filters: altFilters
   })
   expect(wrapper).toMatchSnapshot()
})

test('should handle onDatesChange', () => {
   const startDate = moment(0).add(10, 'days')
   const endDate = moment(0).add(20, 'days')
   wrapper.find('withStyles(DateRangePicker)').prop('onDatesChange')({ startDate, endDate })
   expect(setEndDate).toHaveBeenLastCalledWith(endDate)
   expect(setStartDate).toHaveBeenLastCalledWith(startDate)
})

test('should handle on focusChange', () => {
   const focusedInput = 'startDate'
   wrapper.find('withStyles(DateRangePicker)').prop('onFocusChange')(focusedInput)
   expect(wrapper.state()).toEqual({ focusedInput })
})

test('should handle filterByPlace', () => {
   const value = 'bad'
   wrapper.find('select').at(1).simulate('change', {
      target: { value }
   })
   expect(filterByPlace).toHaveBeenLastCalledWith(value)
})

test('should handle filterByName', () => {
   const value = 'kylu'
   wrapper.find('select').at(2).simulate('change', {
      target: { value }
   })
   expect(filterByName).toHaveBeenLastCalledWith(value)
})

test('should handle the sortDesc', () => {
   const value = 'desc'
   wrapper.find('select').at(0).simulate('change', {
      target: { value }
   })
   expect(sortDesc).toHaveBeenCalledTimes(1)
})



