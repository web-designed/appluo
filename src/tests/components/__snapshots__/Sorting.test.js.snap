// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Sorting correctly with altered filters 1`] = `
<div>
  <h6>
    Filter & Sort
  </h6>
  <div
    class="row"
  >
    <div
      class="input-group col-md-3 col-lg-2 mb-3"
    >
      <div
        class="input-group-prepend"
      >
        <label
          class="input-group-text"
          for="sorting"
        >
          Sort
        </label>
      </div>
      <select
        class="custom-select"
        id="sorting"
        name="sorting"
        onChange={[Function]}
        value="desc"
      >
        <option
          value="asc"
        >
          ASC
        </option>
        <option
          value="desc"
        >
          DESC
        </option>
      </select>
    </div>
    <div
      class="input-group col-lg-3 mb-3"
    >
      <div
        class="input-group-prepend"
      >
        <label
          class="input-group-text"
          for="place"
        >
          Place
        </label>
      </div>
      <select
        class="custom-select"
        id="place"
        name="place"
        onChange={[Function]}
      >
        <option
          value="all"
        >
          All
        </option>
        <option
          key="index"
          selected={false}
          value="all"
        >
          All
        </option>
        <option
          key="index"
          selected={false}
          value="k端che"
        >
          K端che
        </option>
        <option
          key="index"
          selected={false}
          value="bad"
        >
          Bad
        </option>
        <option
          key="index"
          selected={false}
          value="flur"
        >
          Flur
        </option>
      </select>
    </div>
    <div
      class="input-group col-lg-2 mb-3"
    >
      <div
        class="input-group-prepend"
      >
        <label
          class="input-group-text"
          for="cleaners"
        >
          Cleaners
        </label>
      </div>
      <select
        class="custom-select"
        id="cleaners"
        name="cleaner"
        onChange={[Function]}
      >
        <option
          value="all"
        >
          All
        </option>
        <option
          key="0"
          selected={false}
          value="all"
        >
          All
        </option>
        <option
          key="1"
          selected={false}
          value="kylu"
        >
          Kylu
        </option>
        <option
          key="2"
          selected={false}
          value="alex"
        >
          Alex
        </option>
        <option
          key="3"
          selected={false}
          value="till"
        >
          Till
        </option>
        <option
          key="4"
          selected={false}
          value="chris"
        >
          Chris
        </option>
      </select>
    </div>
    <div
      class="input-group col-md-9 col-lg-5 mb-3"
    >
      <div
        class="input-group-prepend"
      >
        <label
          class="input-group-text"
          for="cleaners"
        >
          Date
        </label>
      </div>
      <withStyles(DateRangePicker)
        anchorDirection="left"
        appendToBody={false}
        block={false}
        calendarInfoPosition="bottom"
        customArrowIcon={null}
        customCloseIcon={null}
        customInputIcon={null}
        daySize={39}
        disableScroll={false}
        disabled={false}
        displayFormat={[Function]}
        enableOutsideDays={false}
        endDate={"1970-01-06T00:00:00.000Z"}
        endDateId="endDateId"
        endDatePlaceholderText="End Date"
        firstDayOfWeek={null}
        focusedInput={null}
        hideKeyboardShortcutsPanel={false}
        horizontalMargin={0}
        initialVisibleMonth={null}
        inputIconPosition="before"
        isDayBlocked={[Function]}
        isDayHighlighted={[Function]}
        isOutsideRange={[Function]}
        isRTL={false}
        keepFocusOnInput={false}
        keepOpenOnDateSelect={false}
        minimumNights={1}
        monthFormat="MMMM YYYY"
        navNext={null}
        navPrev={null}
        noBorder={false}
        numberOfMonths={1}
        onClose={[Function]}
        onDatesChange={[Function]}
        onFocusChange={[Function]}
        onNextMonthClick={[Function]}
        onPrevMonthClick={[Function]}
        openDirection="down"
        orientation="horizontal"
        phrases={
          Object {
            "calendarLabel": "Calendar",
            "chooseAvailableEndDate": [Function],
            "chooseAvailableStartDate": [Function],
            "clearDates": "Clear Dates",
            "closeDatePicker": "Close",
            "dateIsSelected": [Function],
            "dateIsUnavailable": [Function],
            "enterKey": "Enter key",
            "escape": "Escape key",
            "focusStartDate": "Interact with the calendar and add the check-in date for your trip.",
            "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
            "homeEnd": "Home and end keys",
            "jumpToNextMonth": "Move forward to switch to the next month.",
            "jumpToPrevMonth": "Move backward to switch to the previous month.",
            "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
            "keyboardShortcuts": "Keyboard Shortcuts",
            "leftArrowRightArrow": "Right and left arrow keys",
            "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
            "moveFocusByOneMonth": "Switch months.",
            "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
            "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
            "openThisPanel": "Open this panel.",
            "pageUpPageDown": "page up and page down keys",
            "questionMark": "Question mark",
            "returnFocusToInput": "Return to the date input field.",
            "selectFocusedDate": "Select the date in focus.",
            "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
            "upArrowDownArrow": "up and down arrow keys",
          }
        }
        readOnly={false}
        regular={false}
        renderCalendarInfo={null}
        renderDayContents={null}
        renderMonthElement={null}
        renderMonthText={null}
        reopenPickerOnClearDates={false}
        required={false}
        screenReaderInputMessage=""
        showClearDates={true}
        showDefaultInputIcon={false}
        small={false}
        startDate={"1970-01-01T00:00:00.000Z"}
        startDateId="startDateId"
        startDatePlaceholderText="Start Date"
        verticalHeight={null}
        verticalSpacing={22}
        weekDayFormat="dd"
        withFullScreenPortal={false}
        withPortal={false}
      />
    </div>
  </div>
</div>
`;

exports[`should render Sorting correctly with default filters 1`] = `
<div>
  <h6>
    Filter & Sort
  </h6>
  <div
    class="row"
  >
    <div
      class="input-group col-md-3 col-lg-2 mb-3"
    >
      <div
        class="input-group-prepend"
      >
        <label
          class="input-group-text"
          for="sorting"
        >
          Sort
        </label>
      </div>
      <select
        class="custom-select"
        id="sorting"
        name="sorting"
        onChange={[Function]}
        value="asc"
      >
        <option
          value="asc"
        >
          ASC
        </option>
        <option
          value="desc"
        >
          DESC
        </option>
      </select>
    </div>
    <div
      class="input-group col-lg-3 mb-3"
    >
      <div
        class="input-group-prepend"
      >
        <label
          class="input-group-text"
          for="place"
        >
          Place
        </label>
      </div>
      <select
        class="custom-select"
        id="place"
        name="place"
        onChange={[Function]}
      >
        <option
          value="all"
        >
          All
        </option>
        <option
          key="index"
          selected={true}
          value="all"
        >
          All
        </option>
        <option
          key="index"
          selected={false}
          value="k端che"
        >
          K端che
        </option>
        <option
          key="index"
          selected={false}
          value="bad"
        >
          Bad
        </option>
        <option
          key="index"
          selected={false}
          value="flur"
        >
          Flur
        </option>
      </select>
    </div>
    <div
      class="input-group col-lg-2 mb-3"
    >
      <div
        class="input-group-prepend"
      >
        <label
          class="input-group-text"
          for="cleaners"
        >
          Cleaners
        </label>
      </div>
      <select
        class="custom-select"
        id="cleaners"
        name="cleaner"
        onChange={[Function]}
      >
        <option
          value="all"
        >
          All
        </option>
        <option
          key="0"
          selected={true}
          value="all"
        >
          All
        </option>
        <option
          key="1"
          selected={false}
          value="kylu"
        >
          Kylu
        </option>
        <option
          key="2"
          selected={false}
          value="alex"
        >
          Alex
        </option>
        <option
          key="3"
          selected={false}
          value="till"
        >
          Till
        </option>
        <option
          key="4"
          selected={false}
          value="chris"
        >
          Chris
        </option>
      </select>
    </div>
    <div
      class="input-group col-md-9 col-lg-5 mb-3"
    >
      <div
        class="input-group-prepend"
      >
        <label
          class="input-group-text"
          for="cleaners"
        >
          Date
        </label>
      </div>
      <withStyles(DateRangePicker)
        anchorDirection="left"
        appendToBody={false}
        block={false}
        calendarInfoPosition="bottom"
        customArrowIcon={null}
        customCloseIcon={null}
        customInputIcon={null}
        daySize={39}
        disableScroll={false}
        disabled={false}
        displayFormat={[Function]}
        enableOutsideDays={false}
        endDate={null}
        endDateId="endDateId"
        endDatePlaceholderText="End Date"
        firstDayOfWeek={null}
        focusedInput={null}
        hideKeyboardShortcutsPanel={false}
        horizontalMargin={0}
        initialVisibleMonth={null}
        inputIconPosition="before"
        isDayBlocked={[Function]}
        isDayHighlighted={[Function]}
        isOutsideRange={[Function]}
        isRTL={false}
        keepFocusOnInput={false}
        keepOpenOnDateSelect={false}
        minimumNights={1}
        monthFormat="MMMM YYYY"
        navNext={null}
        navPrev={null}
        noBorder={false}
        numberOfMonths={1}
        onClose={[Function]}
        onDatesChange={[Function]}
        onFocusChange={[Function]}
        onNextMonthClick={[Function]}
        onPrevMonthClick={[Function]}
        openDirection="down"
        orientation="horizontal"
        phrases={
          Object {
            "calendarLabel": "Calendar",
            "chooseAvailableEndDate": [Function],
            "chooseAvailableStartDate": [Function],
            "clearDates": "Clear Dates",
            "closeDatePicker": "Close",
            "dateIsSelected": [Function],
            "dateIsUnavailable": [Function],
            "enterKey": "Enter key",
            "escape": "Escape key",
            "focusStartDate": "Interact with the calendar and add the check-in date for your trip.",
            "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
            "homeEnd": "Home and end keys",
            "jumpToNextMonth": "Move forward to switch to the next month.",
            "jumpToPrevMonth": "Move backward to switch to the previous month.",
            "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
            "keyboardShortcuts": "Keyboard Shortcuts",
            "leftArrowRightArrow": "Right and left arrow keys",
            "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
            "moveFocusByOneMonth": "Switch months.",
            "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
            "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
            "openThisPanel": "Open this panel.",
            "pageUpPageDown": "page up and page down keys",
            "questionMark": "Question mark",
            "returnFocusToInput": "Return to the date input field.",
            "selectFocusedDate": "Select the date in focus.",
            "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
            "upArrowDownArrow": "up and down arrow keys",
          }
        }
        readOnly={false}
        regular={false}
        renderCalendarInfo={null}
        renderDayContents={null}
        renderMonthElement={null}
        renderMonthText={null}
        reopenPickerOnClearDates={false}
        required={false}
        screenReaderInputMessage=""
        showClearDates={true}
        showDefaultInputIcon={false}
        small={false}
        startDate={null}
        startDateId="startDateId"
        startDatePlaceholderText="Start Date"
        verticalHeight={null}
        verticalSpacing={22}
        weekDayFormat="dd"
        withFullScreenPortal={false}
        withPortal={false}
      />
    </div>
  </div>
</div>
`;
